torch/testing/_internal/common_utils.py

```diff
@@ -341,7 +342,6 @@ def _check_module_exists(name):
 TEST_WITH_ASAN = os.getenv('PYTORCH_TEST_WITH_ASAN', '0') == '1'
 TEST_WITH_TSAN = os.getenv('PYTORCH_TEST_WITH_TSAN', '0') == '1'
 TEST_WITH_UBSAN = os.getenv('PYTORCH_TEST_WITH_UBSAN', '0') == '1'
-TEST_WITH_ROCM = os.getenv('PYTORCH_TEST_WITH_ROCM', '0') == '1'
```

Comment:
I just thought of an alternative idea.  We can keep the env var, for those that might want to override the behavior for whatever reason, but we can just change the default.
```
TEST_WITH_ROCM = os.getenv('PYTORCH_TEST_WITH_ROCM', '1' if torch.version.hip else '0') == '1'
```

Reply:
I must admit I have doubts that that option would be discoverable enough to be used by anyone.

Reply:
We ROCm PyTorchers know the env var exists, and as we work more towards greater unit test parity with CUDA I can see it being useful to keep the env var to see if tests that had previously been skipped are still broken.  My 2 cents.